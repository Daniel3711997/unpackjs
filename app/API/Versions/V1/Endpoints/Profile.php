<?php

declare(strict_types=1);

namespace Unpack\API\Versions\V1\Endpoints;

use WP_REST_Server;
use WP_REST_Response;
use Unpack\API\Versions\V1\API;
use Unpack\Interfaces\API as APIInterface;

class Profile extends API implements APIInterface {
    public string $route = 'profile';

    public function options(): array {
        return [
            'callback' => [$this, 'getProfile'],
            'methods' => WP_REST_Server::READABLE,
            'permission_callback' => '__return_true',
        ];
    }

    public function getProfile(): WP_REST_Response {
        return $this->createResponse([
            'id' => 1234,
            'name' => 'John Doe',
        ]);
    }
}
