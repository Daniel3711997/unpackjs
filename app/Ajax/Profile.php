<?php

/** @noinspection PhpUnusedAliasInspection */

declare(strict_types=1);

namespace Unpack\Ajax;

use Unpack\Interfaces\Ajax;
use Unpack\Annotations\AjaxRoute;

/**
 * @AjaxRoute(name="getProfile", method="getProfileAjax")
 */
class Profile implements Ajax {
    public static function getProfileAjax(): array {
        return [
            'success' => true,
            'data' => [
                'id' => 1234,
                'name' => 'John Doe',
            ]
        ];
    }
}
