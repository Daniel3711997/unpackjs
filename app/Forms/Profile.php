<?php

/** @noinspection PhpUnusedAliasInspection */

declare(strict_types=1);

namespace Unpack\Forms;

use Unpack\Interfaces\Form;
use Unpack\Annotations\FormRoute;

/**
 * @FormRoute(name="getProfile", method="getProfileForm", availability=1)
 */
class Profile implements Form {
    /**
     * @used
     */
    public static function getProfileForm(): array {
        wp_die(
            json_encode([
                'success' => true,
                'data' => [
                    'id' => 1234,
                    'name' => 'John Doe',
                ]
            ])
        );
    }
}
