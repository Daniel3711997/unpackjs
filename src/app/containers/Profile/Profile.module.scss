.black {
    width: clamp(100px, 50%, 200px);
    color: #000;
}

.white {
    color: #fff;
}
