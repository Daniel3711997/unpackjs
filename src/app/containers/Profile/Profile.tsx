import styles from './StylesProfile.module.scss';

export const Profile = () => {
    return (
        <div>
            <p className={styles.black}>Hello World</p>
        </div>
    );
};
