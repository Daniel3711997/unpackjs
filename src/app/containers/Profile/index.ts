import { createContainer } from '@app/helpers/app';

import { Profile } from './Profile';

const renderContainer = createContainer({
    container: Profile,
    element: 'rootProfile',
});

renderContainer();
