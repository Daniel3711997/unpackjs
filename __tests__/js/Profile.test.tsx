import { render } from '@testing-library/react';

import { Profile } from 'app/containers/Profile/Profile';

test('example', () => {
    // prettier-ignore
    const {
        container
    } = render(<Profile />);

    expect(container).toMatchSnapshot();
});
