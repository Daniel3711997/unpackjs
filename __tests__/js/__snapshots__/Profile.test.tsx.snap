// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example 1`] = `
<div>
  <div>
    <p
      class="black"
    >
      Hello World
    </p>
  </div>
</div>
`;
